controller/nodegroup.go

```diff
@@ -0,0 +1,274 @@
+package controller
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/eks"
+	eksv1 "github.com/rancher/eks-operator/pkg/apis/eks.cattle.io/v1"
+	"github.com/rancher/eks-operator/templates"
+	"github.com/rancher/eks-operator/utils"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	launchTemplateNameFormat = "rancher-managed-lt-%s"
+	launchTemplateTagKey     = "rancher-managed-template"
+	launchTemplateTagValue   = "do-not-modify-or-delete"
+	storageDeviceName        = "/dev/sdr"
+)
+
+func createLaunchTemplate(clusterDisplayName string, ec2Service *ec2.EC2) (*eksv1.LaunchTemplate, error) {
+	launchTemplateCreateInput := &ec2.CreateLaunchTemplateInput{
+		LaunchTemplateData: &ec2.RequestLaunchTemplateData{UserData: aws.String("placeholder")},
```

Comment:
We don't want to use the default version (1) of the rancher-managed launch template because it can never be deleted (until the launch template is deleted). Therefore, we create the launch template with version 1 having this placeholder userdata because it can't be blank.

Reply:
Surprised user data can't be blank. So we add a placeholder string, and that would be on every version 1 of every instance-template ? Or only those where user doesn't specify one.

Reply:
When I said, "it can't be blank" I meant the launch template. We want to create a launch template, but we don't want to use version 1 because it can never be deleted (and we want to be able to clean up versions when node groups are deleted/updated). Therefore this first version will not be used. Since we can't create a blank launch template, we create one with fake userdata.

I added a comment in the code to explain this as well.